### PR TITLE
DAML-LF: Simplify inference of output transaction/value version.

### PR DESCRIPTION
**Pull request by remyhaemmerle-da**
As part of #5164, we simplify the way transaction and value versions
are inferred for the engine output.

The versioning of a transaction `tr` is done as follow:
* Let  `tvMin` be the minimal transaction version allowed by the DAML engine configuration
* Let `tvMax` be the maximal transaction version allowed by the DAML engine configuration
* Let `n₁, ..., nₘ` be the nodes of `tr`.
* Let `pkgᵢ` be the package of the template associated to the node `nᵢ`
* Let `lvᵢ` be the maximal language version that `pkgᵢ` uses (directly or through its dependencies)
* let `tvᵢ` be the maximal transaction version supported by all engines that support both `lvᵢ` and `tvMin`
* Let `vvᵢ` be the maximal value version supported by all engines that support `tvᵢ`.
* Let `tv` be the maximal transaction version between `tv₁, ..., tvₘ`.
* If `tv` is not greater than `tvMax`
  + then
    - Version the values of each node `nᵢ` according `vvᵢ`
    - Version `tr` according `tv`
  + fail otherwise     
  

CHANGELOG_BEGIN
CHANGELOG_END

### Pull Request Checklist

- [X] Read and understand the [contribution guidelines](https://github.com/digital-asset/daml/blob/master/CONTRIBUTING.md)
- [X] Include appropriate tests
- [X] Set a descriptive title and thorough description
- [X] Add a reference to the [issue this PR will solve](https://github.com/digital-asset/daml/issues), if appropriate
- [ ] Include changelog additions in one or more commit message bodies between the `CHANGELOG_BEGIN` and `CHANGELOG_END` tags
- [ ] Normal production system change, include purpose of change in description

NOTE: CI is not automatically run on non-members pull-requests for security
reasons. The reviewer will have to comment with `/AzurePipelines run` to
trigger the build.

### CODE REVIEW

**Comment by remyhaemmerle-da** [comment]
block by #6747

**Comment by S11001001** [comment]
I would reword "any engine" as "all engines".

It would be worth describing the new restriction's implications, that it is impossible to employ features in a new value or transaction version that is not accompanied or succeeded by a language version, in `governance.rst`.
